src/SyntaxLookup.res

```diff
@@ -153,75 +153,64 @@ type state =
   | ShowFiltered(string, array<item>) // (search, filteredItems)
   | ShowDetails(item)
 
+type action =
+  | URLChanged(string)
+  | SearchValueChanged(string)
+  | ItemSelected(item)
+
+let stateFromPath = path =>
+  getAnchor(path)
+  ->Belt.Option.flatMap(anchor =>
+    Js.Array2.find(allItems, item => GithubSlugger.slug(item.id) === anchor)
+  )
+  ->Belt.Option.map(item => ShowDetails(item))
+  ->Belt.Option.getWithDefault(ShowAll)
+
 @react.component
 let make = () => {
   let router = Next.Router.useRouter()
-  let (state, setState) = React.useState(_ => ShowAll)
-
-  React.useEffect0(() => {
-    switch getAnchor(router.asPath) {
-    | Some(anchor) =>
-      Js.Array2.find(allItems, item =>
-        GithubSlugger.slug(item.id) === anchor
-      )->Belt.Option.forEach(item => {
-        setState(_ => ShowDetails(item))
-      })
-    | None => ()
-    }
-    None
-  })
-
-  /*
-  This effect syncs the url anchor with the currently shown final match
-  within the fuzzy finder. If there is a new match that doesn't align with
-  the current anchor, the url will be rewritten with the correct anchor.
-
-  In case a selection got removed, it will also remove the anchor from the url.
-  We don't replace on every state change, because replacing the url is expensive
- */
-  React.useEffect1(() => {
-    switch (state, getAnchor(router.asPath)) {
-    | (ShowDetails(item), Some(anchor)) =>
-      let slug = GithubSlugger.slug(item.id)
-
-      if slug !== anchor {
-        router->Next.Router.replace("syntax-lookup#" ++ anchor)
-      } else {
-        ()
-      }
-    | (ShowDetails(item), None) =>
-      router->Next.Router.replace("syntax-lookup#" ++ GithubSlugger.slug(item.id))
-    | (_, Some(_)) => router->Next.Router.replace("syntax-lookup")
-    | _ => ()
-    }
-    None
-  }, [state])
```

Comment:
@ryyppy For me, the improvement is being able to connect a specific action (like clearing the search box) to this effect (updating the URL). With ReactUpdate, you can queue the effect at the same location where you handle the action, whereas in the old version you have to compare the current URL and new state to reverse engineer what the most recent change was.